src/actions/auth.ts

```diff
@@ -70,8 +45,13 @@ export const signOut = async (redirectTo?: string) => {
   redirect(redirectTo ?? "/explorer");
 };
 
-export const getAuthCookie = async () => {
-  const cookieStore = await cookies();
-  const authCookie = cookieStore.get(GUILD_AUTH_COOKIE_NAME);
-  return authCookie && tokenSchema.parse(jwtDecode(authCookie.value));
+export const tryGetToken = async () => {
+  const token = (await cookies()).get(GUILD_AUTH_COOKIE_NAME)?.value;
+  if (!token) throw new Error("Failed to retrieve token");
+  return token;
+};
+
+export const tryGetParsedToken = async () => {
```

Comment:
I'm not sure if this should throw or return undefined instead. Let's talk about it later!